applications/DEMApplication/tests/test_clusters.py

```diff
@@ -19,67 +19,89 @@
 def GetFilePath(fileName):
     return os.path.join(os.path.dirname(os.path.realpath(__file__)), fileName)
 
-class ClustersTestSolution1(KratosMultiphysics.DEMApplication.DEM_analysis_stage.DEMAnalysisStage, KratosUnittest.TestCase):
-
-    def GetProblemNameWithPath(self):
-        return os.path.join(self.main_path, self.DEM_parameters["problem_name"].GetString())
-
-    def Finalize(self):
-        tol = 1e-3
-        for node in self.spheres_model_part.Nodes:
-            if node.Id == 21:
-                velz = node.GetSolutionStepValue(KratosMultiphysics.VELOCITY_Z)
-                self.assertAlmostEqual(velz, -0.3888113323025093, delta=tol)
-            if node.Id == 31:
-                velz = node.GetSolutionStepValue(KratosMultiphysics.VELOCITY_Z)
-                self.assertAlmostEqual(velz, 0.029044273544728355, delta=tol)
-
-        del node
-
-        super().Finalize()
-
-class ClustersTestSolution2(KratosMultiphysics.DEMApplication.DEM_analysis_stage.DEMAnalysisStage, KratosUnittest.TestCase):
-
-    def GetProblemNameWithPath(self):
-        return os.path.join(self.main_path, self.DEM_parameters["problem_name"].GetString())
-
-    def Finalize(self):
-        tol = 1e-3
-        for node in self.spheres_model_part.Nodes:
-            if node.Id == 21:
-                velz = node.GetSolutionStepValue(KratosMultiphysics.VELOCITY_Z)
-                self.assertAlmostEqual(velz, -0.3888113323025093, delta=tol)
-            if node.Id == 31:
-                velz = node.GetSolutionStepValue(KratosMultiphysics.VELOCITY_Z)
-                self.assertAlmostEqual(velz, 0.029044273544728355, delta=tol)
-
-        del node
-
-        super().Finalize()
-
-
 class TestClusters(KratosUnittest.TestCase):
 
     def setUp(self):
         source = os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", "custom_elements", "custom_clusters", "linecluster3D.clu")
         destination = os.path.join(os.path.dirname(os.path.realpath(__file__)), subfolder_name, "linecluster3D.clu")
         copyfile(source, destination)
 
-    @classmethod
     def test_clusters_1(self):
         path = os.path.join(os.path.dirname(os.path.realpath(__file__)), subfolder_name)
         parameters_file_name = os.path.join(path, "ProjectParametersDEM.json")
+        with open(parameters_file_name,'r') as parameter_file:
+            project_parameters = KratosMultiphysics.Parameters(parameter_file.read())
         model = KratosMultiphysics.Model()
+
         with auxiliary_functions_for_tests.controlledExecutionScope(path):
-            auxiliary_functions_for_tests.CreateAndRunStageInSelectedNumberOfOpenMPThreads(ClustersTestSolution1, model, parameters_file_name, 1)
+            modified_dem_analysis_stage = KratosMultiphysics.DEMApplication.DEM_analysis_stage.DEMAnalysisStage(model, project_parameters)
+
+            backup_original_finalize = modified_dem_analysis_stage.Finalize
+            def ModifiedFinalize(*args, **kwargs):
```

Comment:
To the @KratosMultiphysics/dem team: please take a look at how the Finalize method (from the DEMAnalysisStage) is modified here, without using overriding.